### ISSUE TITLE
Ubuntu hosts with ENA do not have an eth0 interface.

### ISSUE THREAD

**Issue by phyllisstein** [aws/amazon-vpc-cni-k8s#171]
👋 Hey there! I wanted to flag some difficulty I was having with this package in my cluster. When built from `master` (to get support for `t3.*`-type nodes), the plugin fails to start with the following log messages:

<details>
	<summary>ipamd.log</summary>

```
2018-09-09T16:35:52Z [INFO] Starting L-IPAMD v1.1.0-15-g7332580-dirty  ...
2018-09-09T16:35:52Z [INFO] Testing communication with server
2018-09-09T16:35:52Z [INFO] Running with Kubernetes cluster version: v1.11. git version: v1.11.2. git tree state: clean. commit: bb9ffb1654d4a729bb4cec18ff088eacc153c239. platform: linux/amd64
2018-09-09T16:35:52Z [INFO] Communication with server successful
2018-09-09T16:35:52Z [INFO] Starting Pod controller
2018-09-09T16:35:52Z [DEBUG] Discovered region: us-east-1
2018-09-09T16:35:52Z [DEBUG] Found avalability zone: us-east-1a
2018-09-09T16:35:52Z [DEBUG] Discovered the instance primary ip address: 172.20.41.188
2018-09-09T16:35:52Z [DEBUG] Found instance-id: i-02588d7baad2abfa1
2018-09-09T16:35:52Z [DEBUG] Found instance-type: t3.large
2018-09-09T16:35:52Z [DEBUG] Found primary interface's mac address: 0a:ee:8b:80:0d:bc
2018-09-09T16:35:52Z [DEBUG] Discovered 1 interfaces.
2018-09-09T16:35:52Z [DEBUG] Found device-number: 0
2018-09-09T16:35:52Z [DEBUG] Found account ID: 053960162378
2018-09-09T16:35:52Z [DEBUG] Found eni: eni-0ef3509ac534a616e
2018-09-09T16:35:52Z [DEBUG] Found eni eni-0ef3509ac534a616e is a primary eni
2018-09-09T16:35:52Z [DEBUG] Found security-group id: sg-0c5e5533ba413b428
2018-09-09T16:35:52Z [DEBUG] Found subnet-id: subnet-090e235ac77027e6b
2018-09-09T16:35:52Z [DEBUG] Found vpc-ipv4-cidr-block: 172.20.0.0/16
2018-09-09T16:35:52Z [DEBUG] Total number of interfaces found: 1
2018-09-09T16:35:52Z [DEBUG] Found eni mac address : 0a:ee:8b:80:0d:bc
2018-09-09T16:35:52Z [DEBUG] Using device number 0 for primary eni: eni-0ef3509ac534a616e
2018-09-09T16:35:52Z [DEBUG] Found eni: eni-0ef3509ac534a616e, mac 0a:ee:8b:80:0d:bc, device 0
2018-09-09T16:35:52Z [DEBUG] Found cidr 172.20.32.0/19 for eni 0a:ee:8b:80:0d:bc
2018-09-09T16:35:52Z [DEBUG] Found ip addresses [172.20.41.188] on eni 0a:ee:8b:80:0d:bc
2018-09-09T16:35:52Z [INFO] Setting up host network
2018-09-09T16:35:52Z [ERROR] Failed to setup host networkfailed to configure eth0 RPF check: open /proc/sys/net/ipv4/conf/eth0/rp_filter: no such file or directory
2018-09-09T16:35:52Z [ERROR] initialization failureipamd init: failed to setup host network: failed to configure eth0 RPF check: open /proc/sys/net/ipv4/conf/eth0/rp_filter: no such file or directory
```

</details>

After a little casual grepping, I stumbled on this in the code: https://github.com/aws/amazon-vpc-cni-k8s/blob/7332580f67a1979ae55a1135190de8a80b127e5c/pkg/networkutils/network.go#L156-L172

When I inspected my host's networking setup, I found that writing to `/proc/sys/net/ipv4/conf/eth0/rp_filter` was bound to fail because the host did not have an `eth0` interface. Instead, because it's using the Enhanced Networking Adapter, its primary interface is `ens5`:

```
ubuntu@ip-172-20-41-188:/var/log/aws-routed-eni$ sudo cat /proc/sys/net/ipv4/conf/default/rp_filter
1
ubuntu@ip-172-20-41-188:/var/log/aws-routed-eni$ sudo cat /proc/sys/net/ipv4/conf/eth0/rp_filter
cat: /proc/sys/net/ipv4/conf/eth0/rp_filter: No such file or directory
ubuntu@ip-172-20-41-188:/var/log/aws-routed-eni$ sudo ifconfig
docker0: flags=4099<UP,BROADCAST,MULTICAST>  mtu 1500
        inet 172.17.0.1  netmask 255.255.0.0  broadcast 172.17.255.255
        ether 02:42:8a:74:6a:03  txqueuelen 0  (Ethernet)
        RX packets 0  bytes 0 (0.0 B)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 0  bytes 0 (0.0 B)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

ens5: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 9001
        inet 172.20.41.188  netmask 255.255.224.0  broadcast 172.20.63.255
        inet6 fe80::8ee:8bff:fe80:dbc  prefixlen 64  scopeid 0x20<link>
        ether 0a:ee:8b:80:0d:bc  txqueuelen 1000  (Ethernet)
        RX packets 374723  bytes 407785616 (407.7 MB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 135653  bytes 22388837 (22.3 MB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

lo: flags=73<UP,LOOPBACK,RUNNING>  mtu 65536
        inet 127.0.0.1  netmask 255.0.0.0
        inet6 ::1  prefixlen 128  scopeid 0x10<host>
        loop  txqueuelen 1000  (Local Loopback)
        RX packets 111095  bytes 33204633 (33.2 MB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 111095  bytes 33204633 (33.2 MB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

ubuntu@ip-172-20-41-188:/var/log/aws-routed-eni$ sudo cat /proc/sys/net/ipv4/conf/ens5/rp_filter
1
```

After naïvely changing any instances of `eth0` in the source to `ens5` and redeploying, the plugin booted and functioned fine. That's a _deeply_ hacky solution, though, and so I was hoping support for ENA might become part of the core code base in the future.

Thanks for considering it, and for all your work on this package!

**Comment by liwenwu-amazon** [comment]
@phyllisstein @fasaxc   Here is a workaround in case you do not need node-port,  you can disable AWS_VPC_CNI_NODE_PORT_SUPPORT and set it to false.

**Comment by liwenwu-amazon** [comment]
@phyllisstein @fasaxc  Here is another way to find out the primary interface
`ip route list | grep default | awk '{print $5} `
Can you check if this command works on your ubuntu?  thanks

**Comment by fasaxc** [comment]
That's a shame that not all AWS instances have the same interface names.  This approach might be a little more robust in case there are any other clauses in the route:
```
ip route list | grep default | grep -E  'dev (\w+)' -o | awk '{print $2}'
```

**Comment by phyllisstein** [comment]
Both those commands run fine and return `ens5` on my ENA-enabled machines.

**Comment by phyllisstein** [comment]
If it's helpful, here's the complete output of `ip route list`:

```
ubuntu@ip-172-20-122-197:~$ ip route list
default via 172.20.96.1 dev ens5
172.17.0.0/16 dev docker0 proto kernel scope link src 172.17.0.1 linkdown
172.20.96.0/19 dev ens5 proto kernel scope link src 172.20.122.197
172.20.98.168 dev eniccc81bf45c5 scope link
172.20.123.145 dev eni585d70903ae scope link
```

**Comment by igoratencompass** [comment]
I would say maybe the smartest thing to do is AWS revert to the old way of naming interfaces by using `net.ifnames=0 biosdevname=0` in grub for all ena instance types.

**Comment by phyllisstein** [comment]
I gave that a shot---setting `GRUB_CMDLINE_LINUX_DEFAULT="net.ifnames=0 biosdevname=0"` and running `sudo update-grub && sudo reboot`---but didn't have much luck in getting an ENA-enabled Ubuntu machine to switch back to `eth0`. I wonder if the ENA kernel module is overriding that setting somehow?

**Comment by phyllisstein** [comment]
Fixed in #196.
